src/app/utils.ts

```diff
@@ -150,7 +150,7 @@ export const merge = <T1 extends {}, T2 extends {}>(thing1: T1, thing2: T2): T1
   ...getCommonProperties(thing1, thing2).reduce((result, property) => ({
     ...result,
     ...(isPlainObject(thing1[property]) && isPlainObject(thing2[property])
-      ? {[property]: merge(thing1[property], thing2[property])}
+      ? {[property]: merge(thing1[property] as object, thing2[property] as object)}
```

Comment:
Complaint was:
```
Argument of type 'T1[keyof T1 & keyof T2]' is not assignable to parameter of type '{}'.
  Type 'T1[string] | T1[number] | T1[symbol]' is not assignable to type '{}'.
    Type 'T1[string]' is not assignable to type '{}'.ts(2345)
```

Reply:
Here's how ChatGPT suggested rewriting the function:
```
// Utility function to get common properties between two objects
const getCommonProperties = <T1 extends {}, T2 extends {}>(obj1: T1, obj2: T2): Array<keyof T1 & keyof T2> =>
  Object.keys(obj1).filter((key) => key in obj2) as Array<keyof T1 & keyof T2>;

// Merges two objects (thing1 and thing2) into a new object
export const merge = <T1 extends {}, T2 extends {}>(thing1: T1, thing2: T2): T1 & T2 => {
  const commonProperties = getCommonProperties(thing1, thing2);

  const mergedObject = commonProperties.reduce((result, property) => {
    const mergedValue = isPlainObject(thing1[property]) && isPlainObject(thing2[property])
      ? merge(thing1[property] as PlainObject, thing2[property] as PlainObject)
      : Array.isArray(thing1[property]) && Array.isArray(thing2[property])
        ? [...thing1[property], ...thing2[property]]
        : {};

    return { ...result, [property]: mergedValue };
  }, {});

  return { ...thing1, ...thing2, ...mergedObject };
};
```

Reply:
the same function exists in ts-utils [here](https://github.com/openstax/typescript-service-template/blob/main/packages/utils/src/misc/merge.ts) which uses 4.6

ts-utils is now a dependency of rex because we're using the user loader, so you can just use the helper function from the library if thats easier